iocore/net/quic/test/test_QUICAckFrameCreator.cc

```diff
@@ -37,41 +37,83 @@ TEST_CASE("QUICAckFrameCreator", "[quic]")
   CHECK(frame == nullptr);
 
   // One packet
-  creator.update(level, 1, true);
+  creator.update(level, 1, 10);
   ack_frame = creator.generate_frame(level, UINT16_MAX, UINT16_MAX);
   frame     = std::static_pointer_cast<QUICAckFrame>(ack_frame);
   CHECK(frame != nullptr);
   CHECK(frame->ack_block_count() == 0);
   CHECK(frame->largest_acknowledged() == 1);
   CHECK(frame->ack_block_section()->first_ack_block() == 0);
+  CHECK(frame->ack_block_section()->begin()->gap() == 0);
 
+  // retry
   ack_frame = creator.generate_frame(level, UINT16_MAX, UINT16_MAX);
   frame     = std::static_pointer_cast<QUICAckFrame>(ack_frame);
   CHECK(frame == nullptr);
 
   // Not sequential
-  creator.update(level, 2, true);
-  creator.update(level, 5, true);
-  creator.update(level, 3, true);
-  creator.update(level, 4, true);
+  creator.update(level, 2, 10);
+  creator.update(level, 5, 10);
+  creator.update(level, 3, 10);
+  creator.update(level, 4, 10);
   ack_frame = creator.generate_frame(level, UINT16_MAX, UINT16_MAX);
   frame     = std::static_pointer_cast<QUICAckFrame>(ack_frame);
   CHECK(frame != nullptr);
   CHECK(frame->ack_block_count() == 0);
   CHECK(frame->largest_acknowledged() == 5);
-  CHECK(frame->ack_block_section()->first_ack_block() == 3);
+  CHECK(frame->ack_block_section()->first_ack_block() == 4);
 
   // Loss
-  creator.update(level, 6, true);
-  creator.update(level, 7, true);
-  creator.update(level, 10, true);
+  creator.update(level, 6, 10);
+  creator.update(level, 7, 10);
+  creator.update(level, 10, 10);
   ack_frame = creator.generate_frame(level, UINT16_MAX, UINT16_MAX);
   frame     = std::static_pointer_cast<QUICAckFrame>(ack_frame);
   CHECK(frame != nullptr);
   CHECK(frame->ack_block_count() == 1);
   CHECK(frame->largest_acknowledged() == 10);
   CHECK(frame->ack_block_section()->first_ack_block() == 0);
   CHECK(frame->ack_block_section()->begin()->gap() == 1);
+
+  // forget
+  creator.forget(7, level);
```

Comment:
Seems like `forgot` is exported only for this test. I think we should create an ack frame and pass it to `on_frame_acked` instead.